components/studio/charts/apps/rstudio/chart/templates/deployment.yaml

```diff
@@ -0,0 +1,74 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.appname }}
+  namespace: {{ .Values.namespace }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      release: {{ $.Release.Name }}
+      app: rstudio
+      project: {{ $.Values.project.slug }}
+      type: app
+      pod: rstudio
+  template:
+    metadata:
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/path: /metrics
+        # prometheus.io/port: "8501"
+      labels:
+        release: {{ $.Release.Name }}
+        app: rstudio
+        project: {{ $.Values.project.slug }}
+        type: app
+        pod: rstudio
+    spec:
+    {{ if .Values.app.gpu.enabled }}
+      affinity:
+        {{ .Values.app.affinity | toYaml | nindent 8 | trim }}
+    {{ end }}
+    {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      containers:
+      - name: rstudio
+        image: rocker/rstudio:latest
+        env:
+          - name: DISABLE_AUTH
+            value: "true"
+        imagePullPolicy: IfNotPresent
+        ports:
+          - containerPort: 8787
```

Comment:
Add a name here (`name: backend` or similar) and you can refer to that name in the service instead of messing around with remembering numbers

Reply:
Also, you probably want to use `.Values.service.targetport` or similar here, as the value otherwise makes no sense.